File: backend/api/pets.py
"""
宠物相关 API 路由
"""

from typing import Optional
from fastapi import APIRouter, HTTPException, Query

from schema.pet import PetResponse, PetListResponse
from service import pet_service

router = APIRouter(prefix="/pets", tags=["宠物"])


@router.get("", response_model=PetListResponse)
async def get_pets(
    type: Optional[str] = Query(None, description="宠物类型: all, dog, cat, rabbit"),
    search: Optional[str] = Query(None, description="搜索关键词")
):
    """
    获取宠物列表
    支持按类型筛选和关键词搜索
    """
    pets = pet_service.get_all_pets(pet_type=type, search=search)
    return PetListResponse(data=pets, total=len(pets))


@router.get("/{pet_id}", response_model=PetResponse)
async def get_pet(pet_id: str):
    """
    获取单个宠物详情
    """
    pet = pet_service.get_pet_by_id(pet_id)
    if not pet:
        raise HTTPException(status_code=404, detail="宠物不存在")
    return pet


@router.get("/stories/list")
async def get_stories():
    """
    获取故事列表
    """
    stories = pet_service.get_all_stories()
    return {"data": stories, "total": len(stories)}

File: api/pets.ts
/**
 * 宠物 API 服务
 */

import { get } from './client';
import type { Pet, Story } from '../types';

interface PetListResponse {
    data: Pet[];
    total: number;
}

interface StoryListResponse {
    data: Story[];
    total: number;
}

/**
 * 获取宠物列表
 * @param type 宠物类型筛选 (all, dog, cat, rabbit)
 * @param search 搜索关键词
 */
export async function fetchPets(
    type?: string,
    search?: string
): Promise<Pet[]> {
    const params = new URLSearchParams();
    if (type && type !== 'all') {
        params.append('type', type);
    }
    if (search) {
        params.append('search', search);
    }

    const queryString = params.toString();
    const endpoint = `/pets${queryString ? `?${queryString}` : ''}`;

    const response = await get<PetListResponse>(endpoint);
    return response.data;
}

/**
 * 获取单个宠物详情
 */
export async function fetchPetById(petId: string): Promise<Pet> {
    return get<Pet>(`/pets/${petId}`);
}

/**
 * 获取故事列表
 */
export async function fetchStories(): Promise<Story[]> {
    const response = await get<StoryListResponse>('/pets/stories/list');
    return response.data;
}
